packages/thumbnail-grid-block/src/components/Thumbnail/Item.tsx

```diff
@@ -13,34 +13,141 @@ import {
-    const [localAltText, setLocalAltText] = useState<string | undefined>(altText);
+export const Item = forwardRef<HTMLDivElement, ThumbnailItemProps>(
+    (
+        {
+            appBridge,
+            item,
+            image,
+            isEditing,
+            transformStyle = {},
+            draggableProps = {},
+            setActivatorNodeRef,
+            isDragging = false,
+            setUploadedId,
+            openAssetChooser,
+            openFileDialog,
+            updateItemWith,
+            onRemoveAsset,
+            thumbnailStyles,
+            isLoading,
+            onFilesDrop,
+            replaceWithPlaceholder = false,
+        },
+        ref
+    ) => {
+        const [showAltTextMenu, setShowAltTextMenu] = useState(false);
+        const { id, title, description, altText } = item;
+        const [localAltText, setLocalAltText] = useState<string | undefined>(altText);
 
-    const onOpenFileDialog = () => {
-        setUploadedId(id);
-        openFileDialog();
-    };
-    const onAssetChooserClick = () => openAssetChooser(id);
+        const onOpenFileDialog = () => {
+            setUploadedId(id);
+            openFileDialog();
+        };
+        const onAssetChooserClick = () => openAssetChooser(id);
+
+        return (
+            <div
+                key={id}
+                ref={ref}
+                style={{ ...transformStyle, ...(!isDragging ? { zIndex: undefined } : {}) }}
+                className={merge([!isEditing && !image && 'tw-hidden', 'tw-w-full tw-h-full tw-bg-white'])}
+            >
+                <BlockItemWrapper
+                    isDragging={isDragging}
+                    shouldFillContainer={true}
+                    shouldHideWrapper={replaceWithPlaceholder || !isEditing}
+                    shouldHideComponent={replaceWithPlaceholder}
+                    shouldBeShown={showAltTextMenu || isDragging}
+                    toolbarFlyoutItems={[
+                        image
+                            ? [
+                                  {
+                                      title: 'Set alt text',
+                                      onClick: () => setShowAltTextMenu(true),
+                                      icon: <IconSpeechBubbleQuote20 />,
+                                  },
+                              ]
+                            : [],
+                        [
+                            {
+                                title: 'Replace with upload',
+                                icon: <IconArrowCircleUp20 />,
+                                onClick: onOpenFileDialog,
+                            },
+                            {
+                                title: 'Replace with asset',
+                                icon: <IconImageStack20 />,
+                                onClick: onAssetChooserClick,
+                            },
+                        ],
+                    ]}
```

Comment:
add filter here `filter((item) => item.length > 0)}` otherwise it looks now like this:

<img width="234" alt="image" src="https://github.com/Frontify/guideline-blocks/assets/42832501/7af818f3-f4ef-4aac-a9fa-2deff490e820">

Reply:
Sorry was removed during refactoring, and missed to re-add it again after